File: File.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>
#include <direct.h>
#include <sys/stat.h>
#include <windows.h>

#define AUTHOR_NAME "MD. Naiem Islam Nahid"
#define BASE_DIR "generated_folders"

// Function to generate random string
void generate_random_word(char *word, int length) {
    const char charset[] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
    int charset_length = strlen(charset);
    
    for (int i = 0; i < length; i++) {
        int index = rand() % charset_length;
        word[i] = charset[index];
    }
    word[length] = '\0';
}

// Function to get current timestamp with nanoseconds
void get_timestamp(char *timestamp) {
    SYSTEMTIME st;
    GetLocalTime(&st);
    
    sprintf(timestamp, "%04d-%02d-%02d_%02d-%02d-%02d-%03d",
            st.wYear, st.wMonth, st.wDay,
            st.wHour, st.wMinute, st.wSecond,
            st.wMilliseconds);
}

// Function to generate UUID-like string
void generate_uuid(char *uuid) {
    const char charset[] = "0123456789abcdef";
    const char* uuid_template = "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx";
    int i;
    
    for (i = 0; uuid_template[i] != '\0'; i++) {
        if (uuid_template[i] == 'x') {
            uuid[i] = charset[rand() % 16];
        } else if (uuid_template[i] == 'y') {
            uuid[i] = charset[(rand() & 0x3) | 0x8];
        } else {
            uuid[i] = uuid_template[i];
        }
    }
    uuid[i] = '\0';
}

// Function to execute git command
void git_commit(const char *message) {
    char command[1024];
    snprintf(command, sizeof(command), "git add . && git commit -m \"%s\" --quiet", message);
    system(command);
}

int main() {
    srand((unsigned int)time(NULL));
    
    // Create base directory
    _mkdir(BASE_DIR);
    
    printf("Starting folder generation process...\n");
    
    // Create 1000 folders
    for (int folder_num = 1; folder_num <= 1000; folder_num++) {
        char random_word[9];
        generate_random_word(random_word, 8);
        
        char folder_name[100];
        snprintf(folder_name, sizeof(folder_name), "%s/%04d_%s", BASE_DIR, folder_num, random_word);
        
        // Create folder
        _mkdir(folder_name);
        
        // Git commit for folder creation
        char commit_msg[200];
        snprintf(commit_msg, sizeof(commit_msg), "Created folder: %04d_%s", folder_num, random_word);
        git_commit(commit_msg);
        
        // Create 100 files in the folder
        for (int file_num = 1; file_num <= 100; file_num++) {
            char timestamp[50];
            get_timestamp(timestamp);
            
            char uuid[37];
            generate_uuid(uuid);
            
            char file_path[256];
            snprintf(file_path, sizeof(file_path), "%s/%04d_%s_%s.txt", 
                    folder_name, folder_num, random_word, timestamp);
            
            // Create and write to file
            FILE *file = fopen(file_path, "w");
            if (file != NULL) {
                fprintf(file, "Timestamp: %s\n", timestamp);
                fprintf(file, "Date: %.10s\n", timestamp);
                fprintf(file, "Created by: %s\n", AUTHOR_NAME);
                fprintf(file, "Folder: %04d_%s\n", folder_num, random_word);
                fprintf(file, "File: %04d_%s_%s.txt\n", folder_num, random_word, timestamp);
                fprintf(file, "UUID: %s\n", uuid);
                fclose(file);
                
                // Git commit for file creation
                snprintf(commit_msg, sizeof(commit_msg), 
                        "Created file in %04d_%s: %04d_%s_%s.txt",
                        folder_num, random_word, folder_num, random_word, timestamp);
                git_commit(commit_msg);
            }
        }
        
        printf("Completed folder %d/1000: %04d_%s\n", 
               folder_num, folder_num, random_word);
    }
    
    printf("Successfully created all folders and files with git commits!\n");
    return 0;
}

File: File.py
import os
import random
import string
import time
from datetime import datetime
import subprocess
import uuid

def generate_random_word(length=8):
    """Generate a random word using letters and numbers."""
    characters = string.ascii_letters + string.digits
    return ''.join(random.choice(characters) for _ in range(length))

def create_folder_structure():
    author_name = "MD. Naiem Islam Nahid"
    base_dir = "generated_folders"
    
    # Create base directory if it doesn't exist
    if not os.path.exists(base_dir):
        os.makedirs(base_dir)
    
    # Create 1000 folders
    for folder_num in range(1, 1001):
        # Generate random folder name with serial number
        random_word = generate_random_word()
        folder_name = f"{folder_num:04d}_{random_word}"
        folder_path = os.path.join(base_dir, folder_name)
        
        # Create folder
        os.makedirs(folder_path)
        
        # Git commit for folder creation
        subprocess.run(['git', 'add', folder_path])
        commit_msg = f"Created folder: {folder_name}"
        subprocess.run(['git', 'commit', '-m', commit_msg])
        
        # Create 100 files in the folder
        for file_num in range(1, 101):
            # Get current timestamp with nanoseconds
            timestamp = datetime.now()
            timestamp_str = timestamp.strftime('%Y-%m-%d_%H-%M-%S-%f')
            
            # Create unique file name
            file_name = f"{folder_name}_{timestamp_str}.txt"
            file_path = os.path.join(folder_path, file_name)
            
            # Write content to file
            with open(file_path, 'w') as f:
                content = f"""Timestamp: {timestamp}
Date: {timestamp.strftime('%Y-%m-%d')}
Created by: {author_name}
Folder: {folder_name}
File: {file_name}
UUID: {uuid.uuid4()}"""
                f.write(content)
            
            # Git commit for file creation
            subprocess.run(['git', 'add', file_path])
            commit_msg = f"Created file in {folder_name}: {file_name}"
            subprocess.run(['git', 'commit', '-m', commit_msg])
        
        print(f"Completed folder {folder_num}/1000: {folder_name}")

if __name__ == "__main__":
    try:
        create_folder_structure()
        print("Successfully created all folders and files with git commits!")
    except Exception as e:
        print(f"An error occurred: {str(e)}")